doc/examples/affine_registration_3d.py

```diff
@@ -0,0 +1,236 @@
+value is 32. We also need to specify the percentage (an integer in (0, 100])of voxels
+to be used for computing the PDFs, the most accurate registration will be obtained by
+using all voxels, but it is also the most time-consuming choice. We specify full
+sampling by passing None instead of an integer
+"""
+
+nbins = 32
+sampling_prop = None
+metric = MattesMIMetric(nbins, sampling_prop)
+
+"""
+To avoid getting stuck at local optima, and to accelerate convergence, we use a
+multi-resolution strategy (similar to ANTS [Avants11]_) by building a Gaussian Pyramid.
+To have as much flexibility as possible, the user can specify how this Gaussian Pyramid
+is built. First of all, we need to specify how many resolutions we want to use. This is
+indirectly specified by just providing a list of the number of iterations we want to
+perform at each resolution. Here we will just specify 3 resolutions and a large number
+of iterations, 10000 at the coarsest resolution, 1000 at the medium resolution and 100
+at the finest. These are the default settings
+"""
+
+level_iters = [10000, 1000, 100]
+
+"""
+To compute the Gaussian pyramid, the original image is first smoothed at each level
+of the pyramid using a Gaussian kernel with the requested sigma. A good initial choice
+is [3.0, 1.0, 0.0], this is the default
+"""
+
+sigmas = [3.0, 1.0, 0.0]
+
+"""
+Now we specify the sub-sampling factors. A good configuration is [4, 2, 1], which means
+that, if the original image shape was (nx, ny, nz) voxels, then the shape of the coarsest
+image will be about (nx//4, ny//4, nz//4), the shape in the middle resolution will be
+about (nx//2, ny//2, nz//2) and the image at the finest scale has the same size as the
+original image. This set of factors is the default
+"""
+
+factors = [4, 2, 1]
+
+"""
+Now we go ahead and instantiate the registration class with the configuration we just
+prepared
+"""
+
+affreg = AffineRegistration(metric=metric,
+                            level_iters=level_iters,
+                            sigmas=sigmas,
+                            factors=factors)
+
+"""
+Using AffineRegistration we can register our images in as many stages as we want, providing
+previous results as initialization for the next (the same logic as in ANTS). The reason why
+it is useful is that registration is a non-convex optimization problem (it may have more
+than one local optima), which means that it is very important to initialize as close to the
+solution as possible. For example, lets start with our (previously computed) rough
+transformation aligning the centers of mass of our images, and then refine it in three
+stages. First look for an optimal translation. The dictionary regtransforms contains all
+available transforms, we obtain one of them by providing its name and the dimension
+(either 2 or 3) of the image we are working with (since we are aligning volumes, the
+dimension is 3)
+"""
+
+transform = regtransforms[('TRANSLATION', 3)]
```

Comment:
I really don't like using this type of dictionary here. 
I think it's much better to write
transform = TranslationTransform3D()